.github/workflows/rack-box-release.yml

```diff
@@ -23,214 +44,90 @@ jobs:
-        include:
-        - builder: docker
-          os: ubuntu-20.04
-          shell: bash
-        - builder: virtualbox
-          os: macos-latest
-          shell: bash
-#        - builder: hyperv
-#          os: windows-latest
-#          shell: msys2 {0}
-
-    steps:
-    - name: Check out RACK source
-      uses: actions/checkout@v2
-      with:
-        repository: ge-high-assurance/RACK
-        ref: ${{ github.event.release.tag_name }}
-        token: ${{ secrets.RACK_CI_PAT }}
-        path: RACK
-
-    - name: Cache Fuseki distribution
-      uses: actions/cache@v2
-      id: cache-fuseki
-      with:
-        path: RACK/rack-box/files/fuseki.tar.gz
-        key: fuseki-3.16.0
-
-    - name: Cache RACK CLI
-      uses: actions/cache@v2
-      id: cache-rack-cli
-      with:
-        path: RACK/rack-box/files/rack.tar.gz
-        key: rack-cli-${{ hashFiles('RACK/RACK-Ontology/**', 'RACK/nodegroups/**') }}
-
-    - name: Cache RACK documentation
-      uses: actions/cache@v2
-      id: cache-rack-doc
-      with:
-        path: |
-          RACK/rack-box/files/documentation.html
-          RACK/rack-box/files/index.html
-        key: rack-doc-${{ hashFiles('RACK.wiki/*.md') }}
-
-    - name: Cache SemTK distribution
-      uses: actions/cache@v2
-      id: cache-semtk
-      with:
-        path: RACK/rack-box/files/semtk.tar.gz
-        key: semtk-2.2.2-SNAPSHOT-v3
-
-    - name: Cache style spreadsheet
-      uses: actions/cache@v2
-      id: cache-style
-      with:
-        path: RACK/rack-box/files/style.css
-        key: style-db9e23f
-
-    - name: Cache systemctl script
-      uses: actions/cache@v2
-      id: cache-systemctl
-      with:
-        path: RACK/rack-box/files/systemctl3.py
-        key: systemctl-1.5.4260
-
-    - name: Cache Ubuntu ISO
-      if: matrix.builder != 'docker'
-      uses: actions/cache@v2
-      id: cache-ubuntu
-      with:
-        path: RACK/rack-box/packer_cache/e30b34e4d3c536763868a42bb9d5b8f14af297b8.iso
-        key: ubuntu-20.04.1
-
-# Doesn't work on windows-latest (nested virtualization is unsupported)
+    # Skip since windows-latest doesn't support nested virtualization yet
     - name: Enable Hyper-V
       if: matrix.builder == 'hyperv' && matrix.os == 'windows-latest'
       shell: powershell
       run: |
         Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
         Add-WindowsFeature RSAT-Hyper-V-Tools -IncludeAllSubFeature
 
-# Doesn't work on ubuntu-20.04 (nested virtualization is unsupported)
+    # Skip since ubuntu-20.04 doesn't support nested virtualization yet
     - name: Install VirtualBox
       if: matrix.builder == 'virtualbox' && matrix.os == 'ubuntu-20.04'
       run: sudo apt-get install -y virtualbox
 
+    - name: Cache Ubuntu ISO
+      if: matrix.builder != 'docker'
+      uses: actions/cache@v2
+      id: cache-ubuntu
+      with:
+        path: RACK/rack-box/packer_cache/e30b34e4d3c536763868a42bb9d5b8f14af297b8.iso
```

Comment:
How big is this ISO? We discovered previously (#200) that the 5GB cache will start evicting the least recently used artifacts as soon as its full, so we should be sure that all of our rarely changing artifacts fit in that 5GB.

Reply:
The ISO takes 914M bytes (in comparison, the semtk tarball, which we're not caching anymore since we're downloading it directly from a GitHub Release page now, takes a whopping 1.8G bytes).  However, I wouldn't be surprised if the ISO is no longer cached by the time we make another VirtualBox image a month from now.  The cache would save some time between a v3.9 release and a v4.0 release a few days later, though.  It's still a good question whether this cache is worth it (all I can say is that once I've run a packer build on my own computer and waited for packer to download the ISO, I'm happy packer caches the ISO in packer_cache to make the next packer build faster).

Regarding whether combining on: push: branches: with on: pull_request: branches: would cause two builds for every push to a branch that has an open PR, I made several pushes to this PR yesterday and I saw only one lint and one rack-box build started for these pushes.  There were two builds, but they weren't redundant builds (we wanted both the lint and the rack-box builds to run).